app/models/health/patient_referral.rb

```diff
@@ -313,8 +307,26 @@ def record_status
       end
     end
 
-    def disenrolled?
-      disenrollment_date.present? || pending_disenrollment_date.present? || removal_acknowledged? || rejected?
+    # In many cases it is handy to consider a pending_disenrollment
+    # if no disenrollment has been yet been recorded
+    def actual_or_pending_disenrollment_date
+      disenrollment_date || pending_disenrollment_date
```

Comment:
Should we cache this? `@actual_or_pending_disenrollment_date ||= disenrollment_date || pending_disenrollment_date` ?  Or, perhaps this isn't expensive enough.  We probably only call it a handful of times per load.

Reply:
Yep, and its to already loaded Date objects so at most a extra couple cycles per call